service/resource_router_test.go

```diff
@@ -0,0 +1,95 @@
+package service
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/kvm-operator/service/resource/configmapv3"
+	"github.com/giantswarm/kvm-operator/service/resource/servicev2"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func Test_Service_newResourceRouter(t *testing.T) {
+	resourceV1 := []framework.Resource{
+		&servicev2.Resource{},
+	}
+	resourceV2 := []framework.Resource{
+		&servicev2.Resource{},
+		&configmapv3.Resource{},
+	}
+
+	versionedResources := map[string][]framework.Resource{
+		"1.0.0": resourceV1,
+		"2.0.0": resourceV2,
+		"":      resourceV1,
+	}
+
+	testCases := []struct {
+		customObject      v1alpha1.KVMConfig
+		expectedResources []framework.Resource
+		errorMatcher      func(err error) bool
```

Comment:
You added an error matcher but it is not used at all for the actual test.

Reply:
done